util/rate_limiter_test.cc

```diff
@@ -416,6 +416,74 @@ TEST_F(RateLimiterTest, LimitChangeTest) {
   ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->DisableProcessing();
 }
 
+TEST_F(RateLimiterTest, AvailableByteSizeExhaustTest) {
+  auto* env = Env::Default();
+  struct Arg {
+    Arg(int32_t _request_size, std::shared_ptr<RateLimiter> _limiter)
+        : request_size(_request_size), limiter(_limiter) {}
+    int32_t request_size;
+    std::shared_ptr<RateLimiter> limiter;
+  };
+
+  auto writer = [](void* p) {
+    auto* arg = static_cast<Arg*>(p);
+    arg->limiter->Request(arg->request_size, Env::IO_USER, nullptr /* stats */,
+                          RateLimiter::OpType::kWrite);
+  };
+
+  int32_t enqueue_count = 0;
+  int64_t total_pending_requests = 0;
```

Comment:
Not sure if we still this variable as the strategy is to look at bytes went through.